src/utils/injectScript.ts

```diff
@@ -0,0 +1,78 @@
+/**
+ * @license Copyright (c) 2003-2024, CKSource Holding sp. z o.o. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+/**
+ * Map of injected scripts. It is used to prevent injecting the same script multiple times.
+ * It happens quite often in React Strict mode when the component is rendered twice.
+ */
+export const INJECTED_SCRIPTS = new Map<string, Promise<void>>();
+
+/**
+ * Injects a script into the document.
+ *
+ * @param src The URL of the script to be injected.
+ * @returns A promise that resolves when the script is loaded.
+ */
+export function injectScript( src: string ): Promise<void> {
+	// Return the promise if the script is already injected by this function.
+	if ( INJECTED_SCRIPTS.has( src ) ) {
+		return INJECTED_SCRIPTS.get( src )!;
+	}
+
+	// Return the promise if the script is already present in the document but not injected by this function.
+	// We are not sure if the script is loaded or not, so we have to show warning in this case.
+	const maybePrevScript = document.querySelector( `script[src="${ src }"]` );
+
+	if ( maybePrevScript ) {
+		console.warn( `Script with "${ src }" src is already present in DOM!` );
+		maybePrevScript.remove();
+	}
+
+	// Inject the script and return the promise.
+	const promise = new Promise<void>( ( resolve, reject ) => {
+		const script = document.createElement( 'script' );
+
+		script.onerror = reject;
+		script.onload = () => {
+			resolve();
+		};
+
+		script.setAttribute( 'injected-by', 'ck-editor-react' );
```

Comment:
```suggestion
		script.setAttribute( 'injected-by', 'ckeditor-integration' );
```

Maybe there's a better name?